dart-impl/base/include/dash/dart/base/mutex.h

```diff
@@ -33,64 +31,105 @@ char __dummy;
 #endif
 } dart_mutex_t;
 
-static inline
+DART_INLINE
 dart_ret_t
 dart__base__mutex_init(dart_mutex_t *mutex)
 {
 #ifdef DART_HAVE_PTHREADS
-  pthread_mutex_init(&mutex->mutex, NULL);
+  // pthread_mutex_init always succeeds
+  pthread_mutex_init(&mutex->mutex,
```

Comment:
Oh boy, my touchpad tricked me into that. Thanks for spotting this, fixed a bunch of other problems as well.